lib/rspec/core/memoized_helpers.rb

```diff
@@ -1,3 +1,5 @@
+require 'thread'
```

Comment:
We generally try to load as little of the standard library as possible.  Consider the case of a user developing a gem that uses multithreading, but they forget to require `'thread'`: their specs would still pass (since RSpec itself is loading `'thread'`), but when they release the gem users of the gem would be likely to hit errors unless something else in their environment was loading `'thread'`.  Loading less of the stdlib also helps keep the boot time of RSpec itself down (although any individual require is unlikely to be noticable, of course).  Ultimately, we don't have an absolute rule about this--we load `optparse` for CLI option parsing since it's so useful, for example--but we always consider the tradeoffs and see if there's a way to avoid loading part of the stdlib.

In this specific case, I understand this was needed for 1.8 in order to make `Mutex` available.  In rspec-mocks, we had a similar situation, and here's how we dealt with it:

https://github.com/rspec/rspec-mocks/blob/v3.2.0/lib/rspec/mocks/space.rb#L134-L146